web.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python2 -u
+
+import logging
+
+from datetime import datetime, timedelta
+
+from flask import Flask, render_template, session, request, redirect, url_for
+
+import door_lib
+from ldap_user.flipdotuser import FlipdotUser
+from ldap_user.webapp import FrontendError
+import ldap_user.config as ldap_config
+import config
+
+logging.basicConfig(level=logging.INFO)
+
+app = Flask(__name__, )
+
+@app.route('/')
+def index():
+    if 'uid' not in session:
+        return render_template('login.html')
+    else:
+        open, open_raw = is_door_open()
+        return render_template('door.html',
+            state="Open" if open else "Closed",
+            state_raw=open_raw)
+
+@app.route('/login', methods=['POST'])
+def login():
+    if request.method == 'POST':
+        uid = request.form.get('uid', '')
+        pwd = request.form.get('password', '')
+        if not uid or not pwd:
+            return redirect("/", 302)
+        try:
+            ldap = FlipdotUser()
+            valid, dn = ldap.login(uid, pwd)
+        except FrontendError as e:
+            return render_template("error.html", message=e.message)
+        if valid:
+            session['uid'] = dn
+            session['user'] = uid
+            session['pass'] = pwd
+        else:
+            session.pop('uid', None)
+        return redirect(url_for('index'))
+    else:
+        return redirect(url_for('index'))
+
+@app.route('/logout')
+def logout():
+    session.clear()
+    return redirect(url_for('index'))
+
+@app.route('/door', methods=['POST'])
+def door():
+    if request.form.get('action', 'closedoor') == 'opendoor':
+
+        if 'uid' not in session:
```

Comment:
without ssl everyone on the network can hijack a session cookie

Reply:
better than sending the pw with every request... but yeah we need ssl for this